include/openssl/ec.h

```diff
@@ -1342,6 +1342,14 @@ void EC_KEY_METHOD_get_verify(const EC_KEY_METHOD *meth,
                                 EVP_PKEY_OP_DERIVE, \
                                 EVP_PKEY_CTRL_GET_EC_KDF_UKM, 0, (void *)(p))
 
+# define EVP_PKEY_CTX_set_sm2_uid(ctx, uid) \
+        EVP_PKEY_CTX_ctrl(ctx, EVP_PKEY_SM2, EVP_PKEY_OP_TYPE_SIG, \
+                                EVP_PKEY_CTRL_SM2_SET_UID, 0, (void*)(uid))
+
+# define EVP_PKEY_CTX_get_sm2_uid(ctx, uid) \
```

Comment:
I will remove the strdup if that's preferable. Personally I prefer to avoid giving out internal pointers that might change over time, as it is easy to create dangling pointers that way if the application hangs on to the result longer than it should.

Reply:
strdup removed